elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/FormulaValidator.java

```diff
@@ -5,95 +5,104 @@
-    }
+                                // Remove args from column
+                                column = argsIndex == -1 ? column : column.substring(0, argsIndex);
+                                // Prefix column with join table name
+                                column = nullOrEmpty(from) ? column : from + PERIOD + column;
+                                // Prefix column with current path
+                                column = nullOrEmpty(path) ? column : path + PERIOD + column;
 
-    @Override
-    protected Void visitFieldDimension(Queryable parent, ColumnProjection dimension) {
-        return null;
+                                return toFormulaReference(column);
+                            }
+                        });
     }
 
-    /**
-     * For a FORMULA column, we just need to check all source columns in the formula expression.
-     *
-     * @param column a column defined with {@link MetricFormula} or {@link DimensionFormula}
-     * @return null
-     */
-    private Void visitFormulaColumn(Queryable source, ColumnProjection column) {
-        if (visited.contains(getColumnId(source, column))) {
-            throw new IllegalArgumentException(referenceLoopMessage(visited, source, column));
-        }
-
-        Type<?> tableClass = dictionary.getEntityClass(source.getName(), source.getVersion());
+    public void validateColumn(Queryable source, ColumnProjection column) {
+        String columnId = getColumnId(source, column);
+        Set<String> visited = new HashSet<>();
+        try {
+            String current = getHandlebarVariables(column.getExpression());
+            visited.add(current);
+            Template template = handlebars.compileInline(current);
 
-        visited.add(getColumnId(source, column));
-        for (String reference : resolveFormulaReferences(column.getExpression())) {
-
-            //Column is from a query instead of a table.  Nothing to validate.
-            if (source != source.getSource()) {
-                continue;
-            } else if (reference.contains(".")) {
-                JoinPath joinToPath = new JoinPath(tableClass, dictionary, reference);
-                Column joinToColumn = getColumn(joinToPath);
-                if (joinToColumn != null) {
-                    visitColumn(joinToColumn.getTable().toQueryable(), joinToColumn.toProjection());
-                }
-            } else {
-                ColumnProjection referenceColumn = source.getColumnProjection(reference);
-
-                // if the reference is to a logical column, check it
-                if (referenceColumn != null && !reference.equals(column.getName())) {
-                    visitColumn(source, referenceColumn);
+            while (!template.collect(TagType.VAR).isEmpty()) {
+                String resolved = template.apply(tableCtx);
+                current = getHandlebarVariables(resolved);
+                if (!visited.add(current)) {
```

Comment:
I'm not sure this is bulletproof.  What if the references are in a different order in the string?  

A better way to validate might iterate over all the columns in a fact table and resolve them - and if the same context object and key come up twice - throw an exception.  This could be implemented by a subclass of the TableContext method that overrides the get call.

Reply:
Implemented using a different approach. With overriding get method in a subclass, I was facing the issue if the expression is `{{name}} {{name}}` then it's not a cycle but same context object and key come up twice.